lib/spack/spack/cmd/install.py

```diff
@@ -132,6 +132,12 @@ def setup_parser(subparser):
     subparser.add_argument(
         '-v', '--verbose', action='store_true',
         help="display verbose build output while installing")
+    subparser.add_argument(
+        '-t', '--terminal-title', action='store_true', dest='terminal_title',
+        default=None, help="update terminal title while installing")
+    subparser.add_argument(
+        '--no-terminal-title', action='store_false', dest='terminal_title',
+        help="do not update terminal title while installing")
```

Comment:
Do we need this? I can imagine `-t` can serve other purposes at some point, and this is going to be effectively the same as `spack -c config:set_terminal_title:false ...` (when that's supported)

Reply:
Do you mean just the short argument or the arguments in general?

Reply:
Or `spack config add config:terminal_title:false` (pending a PR somewhere :p)

Reply:
The feature used to be on by default and @mwkrentel wanted an easy way to turn it off. Now it's off by default and `-t` provides an easy way to turn in on (and in combination with #17229 you can define an alias like `inst` for `install -tv`).

Reply:
Ah okay. Do you think you would actually use that command line option when you also have the config option? To me a command line option only makes sense if there is a need to enable/disable a feature for a single invocation of `spack install`, and I don't see in what circumstances you would enable `terminal_title` just once. But I may be missing something? Running multiple `spack install`'s in the background?

What do others think? My issue is that `-t` is likely to be rarely used, so it takes away a short-hand for [shiny new feature]. And with `--terminal-title` and `--no-terminal-title` I'm just afraid it's going to overwhelm the user when they look at `spack help` :sweat_smile:

Reply:
I could also live without the `install` arguments as I will turn it on all the time anyway. But I will also define an alias for `install -v` as soon as #17229 is merged and would probably include `-t` for good measure.

tl;dr: I don't care about the configuration specifics, I just want to *finally* have useful progress reporting.

Reply:
If you don't introduce the command line args, leave the default to off in config.yaml, then we can probably just merge, since @sethrj already approved and so would I, and @mwkrentel should be happy too as the feature is opt-in.

The bike-shedding about command line args shouldn't be a blocker for the feature, so we can move that to another PR?

Reply:
Sounds good, done!